test/Microsoft.ML.TestFramework/BaseTestBaseline.cs

```diff
@@ -527,7 +500,7 @@ protected bool CheckOutputIsSuffix(string basePath, string outPath, int skip = 0
                         if (line1 == null || line2 == null)
                             Fail("Output and baseline different lengths: '{0}'", relPath);
                         else
-                            Fail(_allowMismatch, "Output and baseline mismatch at line {1}, expected '{2}' but got '{3}' : '{0}'", relPath, count, line1, line2);
```

Comment:
If we are checking for equality, we should always call out mismatches. Especially in this function when we are doing file comparison.